### PR TITLE
Allow setting controller block owner deletion false

### PR DESCRIPTION
**Pull request by nashant**
## What do these changes do?

This will allow setting controller and blockOwnerDeletion from both build_owner_references and append_owner_references

## Description

Previous behaviour:
Error from trying to set multiple owner controllers

New behaviour:
Allow explicitly setting `controller=False`, so no error

Does the change affect end users?
Only if they choose to use it

## Issues/PRs

> Related:
https://github.com/nolar/kopf/issues/687

## Type of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] The code addresses only the mentioned problem, and this problem only
- [x] I think the code is well written
- [X] Unit tests for the changes exist
- [X] Documentation reflects the changes
- [X] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`

### CODE REVIEW

File: kopf/_cogs/structs/bodies.py [hunk 1]
```diff
@@ -246,6 +246,8 @@ def build_object_reference(
 
 def build_owner_reference(
         body: Body,
+        controller: Optional[bool] = True,
+        block_owner_deletion: Optional[bool] = True,
```
Comment (nolar) [inline kopf/_cogs/structs/bodies.py:251]: Can you please make these arguments to be kwarg-only? As it is generally done for all "mode specifiers" in the public interface(s) — unlike for positional arguments which are typically the objects of the action, not its modes.

```suggestion
        *,
        controller: Optional[bool] = True,
        block_owner_deletion: Optional[bool] = True,
```

File: kopf/_kits/hierarchies.py [hunk 2]
```diff
@@ -22,6 +22,8 @@ class _UNSET(enum.Enum):
 def append_owner_reference(
         objs: K8sObjects,
         owner: Optional[bodies.Body] = None,
+        controller: Optional[bool] = True,
+        block_owner_deletion: Optional[bool] = True,
```
Comment (nolar) [inline kopf/_kits/hierarchies.py:27]:
```suggestion
        *,
        controller: Optional[bool] = True,
        block_owner_deletion: Optional[bool] = True,
```

**Review by nolar — CHANGES_REQUESTED** [review]
Would it also make sense to add these same-named arguments to `kopf.adopt()` — an aggregate on top of all these ownership/hierarchy functions? For consistency of the interfaces. Wdyt?

**Comment by nolar** [comment]
Please rebase on `main` and force-push — #859 has fixed the tests.

**Comment by nashant** [comment]
> Would it also make sense to add these same-named arguments to `kopf.adopt()` — an aggregate on top of all these ownership/hierarchy functions? For consistency of the interfaces. Wdyt?

I considered that and decided probably not. As per the comment you left on the issue I referenced I think adopt means that the child is under full control of the parent. If I'm adding an owner ref explicitly then I may well want to have multiple, if I'm adopting then I want that as the only owner

**Comment by nashant** [comment]
I've rebased against main but am still getting the `tests/hierarchies/test_labelling.py` failures

**Comment by nashant** [comment]
Ok, I've re-rebased against your main (forgot I had accidentally pushed to my main) and commits are now clean, but still the same test failures

**Comment by nashant** [comment]
> > Would it also make sense to add these same-named arguments to `kopf.adopt()` — an aggregate on top of all these ownership/hierarchy functions? For consistency of the interfaces. Wdyt?
> 
> I considered that and decided probably not. As per the comment you left on the issue I referenced I think adopt means that the child is under full control of the parent. If I'm adding an owner ref explicitly then I may well want to have multiple, if I'm adopting then I want that as the only owner

@nolar what would you prefer?

**Pull request merged**